### PR TITLE
Fix pending onended and update documentation

### PR DESCRIPTION
**Pull request by ctwomblyamzn**
*Issue #, if available:*

*Description of changes:*
* onEnded was getting triggered because of [this bit of code](https://github.com/amazon-connect/amazon-connect-streams/blob/master/src/core.js#L972). I added pending as an active state to rectify that behavior.
* Updated Documentation.md to reflect the actual behavior of some event handler subscriber APIs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Documentation.md [hunk 1]
```diff
@@ -483,42 +483,39 @@ Subscribe a method to be invoked whenever the contact is updated.
...
-to an API call when it succeeds, and this is usually triggered by a UI interaction such as clicking an
-accept button.  The proper response to this API is to stop playing ringtones and remove any Accept UI buttons
-or actions, and potentially show an "Accepting..." UI to the customer.
+Subscribe a method to be invoked whenever the contact is accepted.
 
 ### `contact.onMissed()`
 ```
 contact.onMissed(function(contact) { ... });
 ```
-Subscribe a method to be invoked whenever the contact is missed. This is an event which is fired when a contact is put in state "missed" by the backend, which happens when the agent does not answer for a certain amount of time, or when the customer hangs up before the agent can answer.
```
Comment (haochi-aws) [inline Documentation.md]: which definition of client is this here? browser, the the streams client, the person on the other end of the contact/connection?

File: Documentation.md [hunk 2]
```diff
@@ -483,42 +483,39 @@ Subscribe a method to be invoked whenever the contact is updated.
...
 ```
-Subscribe a method to be invoked when the contact is pending. Pending occurs after incoming and before connecting, when an agent has accepted or answered, but the backend has not yet acknowledged that a connection is being formed.
+Subscribe a method to be invoked when the contact is pending. This event is expected to occur before the connecting event.
 
 ### `contact.onConnecting()`
 ```
 contact.onConnecting(function (contact) { ... });
 ```
-Subscribe a method to be invoked when the contact is connecting. This works with chat and softphone contacts. This state happens after the agent has accepted a softphone call or answered if not. 
+Subscribe a method to be invoked when the contact is connecting. This works with chat and softphone contacts. This event happens when a call or chat comes in, before accepting (there is an exception for queue callbacks, in which onConnecting's handler is executed happens after the callback is accepted).
```
Comment (haochi-aws) [inline Documentation.md]: probably want to highlight that `accepted` is referencing the `onAccepted` method

File: Documentation.md (hunk 1 above)
Comment (ctwomblyamzn) [reply Documentation.md]: The person talking with the agent. I suppose client is a misleading word. Customer is also misleading... non-agent participant maybe?

File: Documentation.md [hunk 3]
```diff
@@ -483,42 +483,39 @@ Subscribe a method to be invoked whenever the contact is updated.
...
 ```
-Subscribe a method to be invoked when the contact is pending. Pending occurs after incoming and before connecting, when an agent has accepted or answered, but the backend has not yet acknowledged that a connection is being formed.
+Subscribe a method to be invoked when the contact is pending. This event is expected to occur before the connecting event.
 
 ### `contact.onConnecting()`
 ```
 contact.onConnecting(function (contact) { ... });
 ```
-Subscribe a method to be invoked when the contact is connecting. This works with chat and softphone contacts. This state happens after the agent has accepted a softphone call or answered if not. 
+Subscribe a method to be invoked when the contact is connecting. This works with chat and softphone contacts. This event happens when a call or chat comes in, before accepting (there is an exception for queue callbacks, in which onConnecting's handler is executed happens after the callback is accepted). Note that once the contact has been accepted, the `onAccepted` handler will be triggered.
```
Comment (scoland) [inline Documentation.md]: I think there's a small typo here, should be "in which onConnecting's handler is executed after", not "happens after".

**Pull request merged**